autosubmit_api/components/jobs/utils.py

```diff
@@ -144,8 +145,9 @@ def get_job_total_stats(status_code: int, name: str, tmp_path: str) -> Tuple[dat
     current_status = status_from_job
     path = os.path.join(tmp_path, name + '_TOTAL_STATS')
     if os.path.exists(path):
-        request = 'tail -1 ' + path
-        last_line = os.popen(request).readline()
+        process = subprocess.Popen(['tail', '-1', path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, _ = process.communicate()
+        last_line = stdout.decode()
```

Comment:
Agree about using `text=True`. I didn't see that parameter but it will save us the extra `.decode()` we have to do after.

> check_output will [call run](https://github.com/python/cpython/blob/ad9d059eb10ef132edd73075fa6d8d96d95b8701/Lib/subprocess.py#L472-L473) which does all the [heavy-lifting](https://github.com/python/cpython/blob/ad9d059eb10ef132edd73075fa6d8d96d95b8701/Lib/subprocess.py#L554-L579) to execute the command+args and also handles timeout/decoding/errors/etc. using Popen too.

+1. It seems to be compatible with our use case and provides us with extra things.